src/plugin/callbacks.c

```diff
@@ -226,3 +226,12 @@ purple_events_callback_account_signed_on(PurpleConnection *conn, PurpleEventsCon
 
     data->handle = purple_timeout_add_seconds(5, _purple_events_callback_account_signed_on_timeout, data);
 }
+
+void
+purple_events_callback_auth_requested(PurpleAccount *account, const char *remote_user, const char *message, PurpleEventsContext *context)
+{
+    if ( ! purple_events_utils_check_event_dispatch(context, account, NULL, "authorization-requested") )
```

Comment:
The event has a sender, please do as the other events and check `purple_events_utils_check_buddy_event_dispatch()` too.

Reply:
It has a sender, but not a buddy : at the time this event arrives, the sender is not yet in the receiver's buddy list and thus has no associated `PurpleBuddy` structure in the `PurpleAccount`.

Reply:
Fair enough.
Are there protocols needing an authorization ping pong? Like, you add someone, they authorize you and add you, then you have to authorize them back? You probably won’t get the time to block the event anyway… Let’s keep it simple.

Reply:
The only protocol plugins present in the libpurple codebase that use `purple_account_request_authorization` are Jabber and OSCAR. None of those use ping-pong authorisation. Personally, my own motivation for these changes was the skypeweb plugin, which doesn't use ping-pong either. I think it's safe to assume that such a scheme isn't used nowadays...